test/requests/definition_expectations_test.rb

```diff
@@ -344,6 +344,91 @@ def bar
     end
   end
 
+  def test_jumping_to_autoload_definition_when_declaration_exists
+    source = <<~RUBY
+      # typed: false
+
+      class Foo
+        autoload :Bar, "bar"
+
+        class Bar; end
+      end
+    RUBY
+
+    with_server(source) do |server, uri|
+      server.process_message(
+        id: 1,
+        method: "textDocument/definition",
+        params: { textDocument: { uri: uri }, position: { character: 3, line: 3 } },
+      )
+      assert_equal(uri.to_s, server.pop_response.response.first.attributes[:uri])
+    end
+  end
+
+  def test_jumping_to_autoload_definition_with_two_definition
```

Comment:
Multiple declarations for the same constant is already handled and tested by the index, so I'd remove this test.